radis/io/tools.py

```diff
@@ -97,7 +99,7 @@ def _ndarray2df(data, columns, linereturnformat):
     dtype = list(zip(list(columns.keys()), newtype)) + [
         ("_linereturn", linereturnformat)
     ]
-    data = _cast_to_dtype(data, dtype)
+    data = _cast_to_dtype(data, dtype, is_geisa=is_geisa)
```

Comment:
cant you cast to a dtype where dtype is `str` for columns you expect will contain "D", then use a df.column.str.replace() on the DataFrame (much faster than the python loop) ?

Reply:
Yeah, this sounds good. I will try this approach in the next commit.